src/compiler/checker.ts

```diff
@@ -6872,6 +6872,17 @@ namespace ts {
 
         function getPrimitiveTypeFromJSDocTypeReference(node: TypeReferenceNode): Type {

```

Comment:
This isn't really an apropos name anymore. Perhaps call this `getIntendedTypeFromJsDocTypeReference`.